### DISCUSSION TITLE
Handling CVSS environmental score analysis

### DISCUSSION THREAD

**Discussion by fabiolanza** [DependencyTrack/dependency-track#2178] [Q&A]
Hi, during the vulnerability analysis sometimes it's important to calculate the environmental risk/severity score. As an outcome, a vulnerability might drop from CRITICAL to HIGH for example. Is there a way that this can be managed in Dependency-Track?
If there is not a standard way, is it possible to tweak Dependency-Track configuration to accommodate it? For example, one could manually change the severity of a CVE, or add a new field for tracking CVE severity score after evaluation of the intended operational environment.

**Comment by stevespringett** [comment]
This can currently only be achieved for internally defined vulnerabilities. I think keeping the severity the same, but adding a new capability of supporting CVSS and OWASP risk rating scores from multiple sources would be ideal. This would align to the "rating" features in CycloneDX. See https://cyclonedx.org/docs/1.4/json/#vulnerabilities_items_ratings

**Reply by fabiolanza** [reply]
Hi Steve, thanks for the answer. I feel this might lead to something that we can work with, but I quite did not picture how to do what you proposed. Can you please be more specific? Thanks.

**Comment by karsten-klein** [comment]
You may find this interesting: https://www.openchainproject.org/webinar/2024/03/22/webinar-universal-cvss-calculator

Of course there are a whole lot of details in order to enable rescoring at scale.
